src/server/tiered_storage.cc

```diff
@@ -390,6 +402,45 @@ void TieredStorage::FinishIoRequest(int io_res, InflightWriteRequest* req) {
   VLOG_IF(2, num_active_requests_ == 0) << "Finished active requests";
 }
 
+PrimeIterator TieredStorage::Load(DbIndex db_index, PrimeIterator it, string_view key) {
+  PrimeValue* entry = &it->second;
+  CHECK(entry->IsExternal());
+  DCHECK_EQ(entry->ObjType(), OBJ_STRING);
+  auto [offset, size] = entry->GetExternalSlice();
+  string res;
+  res.resize(size);
+  auto ec = Read(offset, size, res.data());
+  CHECK(!ec) << "TBD";
+
+  // Read will preempt, check if iterator still points to our entry
+  PrimeTable* pt = db_slice_.GetTables(db_index).first;
+  if (!it.is_occupied() || it->first != key) {
```

Comment:
Is it still safe to use `it` here?

Reply:
We know we can still access the table as it was not destroyed because we are using read lock
so we are checking inside is_occupied if the segment is still valid (no shrink in dashtable)  and if the the bucket and slots are still used
So yes it is safe